src/api/component.js

```diff
@@ -186,4 +179,12 @@ export default class Component extends HTMLElement {
 
     return Ctor;
   }
+
+  static shouldRender (elem, prev, curr) {
+    for (let name in prev) {
```

Comment:
Do we need to compare `Object.keys(...).length`? what if curr got a new element, is that possible?